lib/exhal/navigation.ex

```diff
@@ -42,7 +42,10 @@ defmodule ExHal.Navigation do
 
     case ExHal.get_links_lazy(a_doc, name, fn -> :missing end) do
       :missing -> missing_link_handler.(name)
-      links -> Enum.map(links, &_follow_link(a_doc.client, &1, tmpl_vars, opts))
+      links ->
+        links
+        |> Task.async_stream(&_follow_link(a_doc.client, &1, tmpl_vars, opts))
+        |> Enum.map(fn {:ok, followed_link} -> followed_link end)
```

Comment:
`_follow_link/4` can fail. What happens if fetching one of the links is unable to connect, times out, returns a 5XX, returns 4XX, returns a non-HAL response?

Reply:
The underlying `@client_module.get/3` returns either `{:ok, _}` or `{:error, _}` which will just be unwrapped by the `Enum.map/2`. Should we handle the possibility of it crashing?

Reply:
Hmm... do we even need the map on line 48? The previous implementation didn't unbox the `_follow_link` return values and the function docs indicate it returns a collection of `:ok`/`:error` tuples.

Reply:
`Task.async_stream` wraps responses in `{:ok, value}` tuples, so we're just unwrapping that.